cpp/src/arrow/util/bitmap_ops.cc

```diff
@@ -255,53 +255,71 @@ void AlignedBitmapOp(const uint8_t* left, int64_t left_offset, const uint8_t* ri
 }
 
 template <template <typename> class BitOp>
-void UnalignedBitmapOp(const uint8_t* left, int64_t left_offset, const uint8_t* right,
-                       int64_t right_offset, uint8_t* out, int64_t out_offset,
-                       int64_t length) {
+int64_t UnalignedBitmapOp(const uint8_t* left, int64_t left_offset, const uint8_t* right,
+                          int64_t right_offset, uint8_t* out, int64_t out_offset,
+                          int64_t length) {
   BitOp<uint64_t> op_word;
   BitOp<uint8_t> op_byte;
 
+  uint64_t out_pop_count = 0;
   auto left_reader = internal::BitmapWordReader<uint64_t>(left, left_offset, length);
   auto right_reader = internal::BitmapWordReader<uint64_t>(right, right_offset, length);
   auto writer = internal::BitmapWordWriter<uint64_t>(out, out_offset, length);
 
   auto nwords = left_reader.words();
   while (nwords--) {
-    writer.PutNextWord(op_word(left_reader.NextWord(), right_reader.NextWord()));
+    auto outWord = op_word(left_reader.NextWord(), right_reader.NextWord());
+    out_pop_count += BitUtil::PopCount(outWord);
+    writer.PutNextWord(outWord);
   }
   auto nbytes = left_reader.trailing_bytes();
   while (nbytes--) {
     int left_valid_bits, right_valid_bits;
     uint8_t left_byte = left_reader.NextTrailingByte(left_valid_bits);
     uint8_t right_byte = right_reader.NextTrailingByte(right_valid_bits);
     DCHECK_EQ(left_valid_bits, right_valid_bits);
-    writer.PutNextTrailingByte(op_byte(left_byte, right_byte), left_valid_bits);
+    auto out_byte = op_byte(left_byte, right_byte);
+    out_pop_count += BitUtil::kBytePopcount[out_byte];
+    writer.PutNextTrailingByte(out_byte, left_valid_bits);
   }
+  return out_pop_count;
 }
 
 template <template <typename> class BitOp>
 void BitmapOp(const uint8_t* left, int64_t left_offset, const uint8_t* right,
-              int64_t right_offset, int64_t length, int64_t out_offset, uint8_t* dest) {
+              int64_t right_offset, int64_t length, int64_t out_offset, uint8_t* dest,
+              int64_t* out_validity_count) {
   if ((out_offset % 8 == left_offset % 8) && (out_offset % 8 == right_offset % 8)) {
     // Fast case: can use bytewise AND
+    // return AlignedBitmapOp<BitOp>(left, left_offset, right, right_offset, dest,
+    // out_offset,
```

Comment:
nit: commented code?

Reply:
It was there from earlier, I wasn't sure if I should remove it. Hence, left it as it is.

Reply:
I don't see it in the original file: https://github.com/apache/arrow/blob/23d19ce754faa32878bd5e48f56e7510d34f15c6/cpp/src/arrow/util/bitmap_ops.cc

Reply:
hm, split memory. May be I only added it then. Sorry, removed it.